jhotdraw-core/src/main/java/org/jhotdraw/draw/figure/BezierFigure.java

```diff
@@ -171,11 +171,12 @@ protected void drawFill(Graphics2D g) {
 
   @Override
   public boolean contains(Point2D.Double p, double scaleDenominator) {
+    boolean result = false;
     double tolerance =
         Math.max(1f, 2 * AttributeKeys.getPerpendicularHitGrowth(this, scaleDenominator));
     if (isClosed() || attr().get(FILL_COLOR) != null && attr().get(UNCLOSED_PATH_FILLED)) {
```

Comment:
This conditional is very complex. Doing the Decompose Conditional refactoring could help understanding what is going on. Basically, write a private method with a proper name which returns a boolean value (the conditional).